code/ship/ship.cpp

```diff
@@ -16977,6 +16980,7 @@ void object_jettison_cargo(object *objp, object *cargo_objp, float jettison_spee
 
 		// set for relative separation speed (see also do_dying_undock_physics)
 		pos = docker_p0_norm;
+		vm_vec_add2(&pos, &cargo_objp->pos);
```

Comment:
I noticed one of my own bugs here.  The whack is supposed to come from the dockpoint global coordinates, so you should actually add `&objp->pos`.

Reply:
Surely you mean the dockpoint's position, instead of the ship's position? Although after having looked for a bit, it's not clear to me how to get that...

Reply:
You're right.  I guess what needs to be done is modify `find_adjusted_dockpoint_normal` (which is only used in this one place) to get the dockpoint itself instead of the normal.

If you need help on rewriting it into `find_adjusted_dockpoint`, I can take a look at that later.

Reply:
It looks like `find_adjusted_dockpoint_info` returns the dock points position, although it gives two positions which are very close together and simply averaged in `dock_orient_and_approach`. It seems like this might be I'm looking for?

Reply:
I based `find_adjusted_dockpoint_normal` on `find_adjusted_dockpoint_info`.  That's why the comment says "// pruned version of below function".

 That being said, computers are faster than they were 15 years ago, and the jettison code will run only once per jettison instance, so you could delete `find_adjusted_dockpoint_normal` and just use the other function.

Reply:
Oh, I see, that's the only place that function is used. Sounds good then.